web-console/src/lib/components/streaming/import/SQLValueInput.tsx

```diff
@@ -14,83 +16,94 @@ import { TextField, TextFieldProps } from '@mui/material'
-            type: 'string'
-          }))
-          .with(SqlType.TIMESTAMP, () => ({
-            type: 'datetime-local'
-          }))
-          .with(SqlType.DATE, () => ({
-            type: 'date'
-          }))
-          .with(SqlType.ARRAY, () => ({
-            type: 'string'
-          }))
-          .otherwise(() => ({
-            type: 'string'
-          }))}
-        {...props}
-      />
-    ))
+>) => {
+  const numericOnChange = {
+    onChange: (event: any) => {
+      if (event.target.value === '' || event.target.value === undefined) {
+        return props.onChange({ ...event, target: { ...event.target, value: null as any } })
+      }
+      return props.onChange(event)
+    }
+  }
+  return (
+    <TextField
+      InputProps={{
+        endAdornment: columnType.nullable ? (
+          <IconButton
+            size='small'
+            sx={{ mr: -3 }}
+            onClick={() => props.onChange({ target: { value: null as any } } as any)}
+          >
+            <IconX></IconX>
+          </IconButton>
+        ) : undefined
+      }}
+      {...match(columnType.type)
+        .with(SqlType.TINYINT, SqlType.SMALLINT, SqlType.INTEGER, SqlType.REAL, SqlType.DOUBLE, () => ({
+          type: 'number',
+          inputProps: {
+            ...(({ min, max }) => ({ min: min.toNumber(), max: max.toNumber() }))(numericRange(columnType)),
+            value: props.value === null ? '' : props.value, // Enable clearing of the input when setting value to null
+            placeholder: props.value === null ? 'null' : ''
+          },
+          ...props,
+          ...numericOnChange
+        }))
+        .with(SqlType.BIGINT, SqlType.DECIMAL, () => ({
+          ...bigNumberInputProps({
+            ...props,
+            ...numericOnChange,
+            defaultValue: props.defaultValue as BigNumber | undefined,
+            precision: columnType.precision,
+            scale: columnType.scale
+          }),
+          placeholder: props.value === null ? 'null' : '',
+          ...props,
+          onChange: undefined
+        }))
+        .with(SqlType.BOOLEAN, () => ({
+          type: 'checkbox',
+          ...props
+        }))
+        .with(SqlType.CHAR, () => ({
+          type: 'string',
+          inputProps: {
+            maxLength: 1
+          },
+          ...props
+        }))
+        .with(SqlType.VARCHAR, () => ({
+          type: 'string',
+          inputProps: {
+            maxLength: columnType.precision ?? 0 > 0 ? columnType.precision : undefined
+          },
+          ...props,
+          value: props.value === null ? '' : props.value,
+          placeholder: props.value === null ? 'null' : ''
+        }))
+        .with(SqlType.TIME, () => ({
+          type: 'string',
+          ...props
+        }))
+        .with(SqlType.TIMESTAMP, () => ({
+          type: 'datetime-local',
+          ...props
+        }))
+        .with(SqlType.DATE, () => ({
+          type: 'date',
```

Comment:
can't this just use some tostring method of the type?

Reply:
Not sure what do you mean? Now the date value is temporarily input as a string value, and is getting converted to a date JS type (via a provided method/function) elsewhere.